core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -885,6 +910,53 @@ private void unparseRegexpExtractAll(SqlWriter writer, SqlCall call,
     writer.endFunCall(regexpExtractAllFrame);
   }
 
+  private void unparseCot(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+    final SqlWriter.Frame cotFrame = writer.startFunCall("1/tan");
```

Comment:
1/tan are two operands and one operator (/) so create SqlCall for 1 and tan and use operator divide.